### PR TITLE
demo: platform options 

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- include parent key in migration logic
- init `platformOptions`

## Context
Ref: https://github.com/RahulGautamSingh/renovate/issues/21

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository https://bitbucket.org/rahulgautamsingh/old-dev-branch/src/main/

### CODE REVIEW

File: lib/config/migrations/custom/bb-use-default-reviewers-migration.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import is from '@sindresorhus/is';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class BbUseDefaultReviewersMigration extends AbstractMigration {
+  override readonly propertyName = 'bbUseDefaultReviewers';
+
+  override run(value: unknown, key: string, parentKey?: string): void {
+    const platformOptions = this.get('platformOptions') ?? {};
```
Comment (viceice) [inline lib/config/migrations/custom/bb-use-default-reviewers-migration.ts]: move to condition, we only need to get the value if the condition is true (perf)

File: lib/config/migrations/custom/bb-use-development-branch-migration.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import is from '@sindresorhus/is';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class BbUseDevelopmentBranchMigration extends AbstractMigration {
+  override readonly propertyName = 'bbUseDevelopmentBranch';
+
+  override run(value: unknown, key: string, parentKey?: string): void {
+    const platformOptions = this.get('platformOptions') ?? {};
```
Comment (viceice) [inline lib/config/migrations/custom/bb-use-development-branch-migration.ts]: same here

File: lib/config/migrations/custom/bb-use-default-reviewers-migration.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    if (is.boolean(value) && parentKey !== 'platformOptions') {
+      this.delete();
+      this.setHard('platformOptions', {
```
Comment (viceice) [inline lib/config/migrations/custom/bb-use-default-reviewers-migration.ts:11]: wouldn't thise move the `platformOptions` key if already exist in config?

File: lib/config/types.ts [hunk 4]
```diff
@@ -300,6 +300,7 @@ export interface RenovateConfig
   statusCheckNames?: Record<StatusCheckKey, string | null>;
   env?: UserEnv;
   logLevelRemap?: LogLevelRemap[];
+  platformOptions?: Record<string, boolean>;
```
Comment (viceice) [inline lib/config/types.ts]: We should create and interface and axplicit properties

File: lib/modules/platform/types.ts [hunk 5]
```diff
@@ -48,8 +48,11 @@ export interface RepoParams {
   renovateUsername?: string;
   cloneSubmodules?: boolean;
   ignorePrAuthor?: boolean;
-  bbUseDevelopmentBranch?: boolean;
+  // bbUseDevelopmentBranch?: boolean;
```
Comment (viceice) [inline lib/modules/platform/types.ts]:
```suggestion
```

File: lib/modules/platform/types.ts [hunk 6]
```diff
@@ -48,8 +48,11 @@ export interface RepoParams {
...
   includeMirrors?: boolean;
+  platformOptions?: {
+    bbUseDevelopmentBranch?: boolean;
+  };
```
Comment (viceice) [inline lib/modules/platform/types.ts]: reuse the mentioned interface

File: lib/config/options/index.ts [hunk 7]
```diff
@@ -3126,6 +3128,13 @@ const options: RenovateOptions[] = [
     default: false,
     globalOnly: true,
   },
+  {
+    name: 'platformOptions',
```
Comment (viceice) [inline lib/config/options/index.ts:2949]: needs docs update

File: lib/config/migrations/types.ts [hunk 8]
```diff
@@ -9,5 +9,5 @@ export interface MigrationConstructor {
 export interface Migration {
   readonly deprecated: boolean;
   readonly propertyName: string | RegExp;
-  run(value: unknown, key: string): void;
+  run(value: unknown, key: string, parentKey?: string): void;
```
Comment (viceice) [inline lib/config/migrations/types.ts]: we should do this in separate PR

File: lib/config/migrations/custom/bb-use-default-reviewers-migration.ts (hunk 3 above)
Comment (RahulGautamSingh) [reply lib/config/migrations/custom/bb-use-default-reviewers-migration.ts:11]: No, it will update it only.

Comment (RahulGautamSingh) [reply lib/config/migrations/custom/bb-use-default-reviewers-migration.ts:11]: yes, but since its a new config option, that should not happen as long as we migrated all the options in one pr.

File: lib/config/migrations/types.ts (hunk 8 above)
Comment (RahulGautamSingh) [reply lib/config/migrations/types.ts]: https://github.com/renovatebot/renovate/pull/30364

**Comment by RahulGautamSingh** [comment]
todo: add platformVersion
